packages/autocomplete-tools/@types/fig/index.d.ts

```diff
@@ -3,25 +3,106 @@ export {};
 
 declare global {
   export namespace Fig {
-    // All the available templates
     type TemplateStrings = "filepaths" | "folders" | "history";
 
+    /**
+     * A template which is a single TemplateString or an array of TemplateStrings.
+     *
+     * @remark
+     * Templates are generators prebuilt by Fig. Here are the three templates:
+     * * filepaths: show folders and filepaths. Allow autoexecute on filepaths
+     * * folders: show folders only. Allow autoexecute on folders
+     * * history: show suggestions for all items in history matching this pattern
+     *
+     * @example
+     * `cd` uses the "folders" template
+     * `ls` useed  ["filepaths", "folders"]. Why both? Because if I `ls` a directory, we want to enable a user to autoexecute on this directory. If we just did "filepaths" they couldn't autoexecute.
+     *
+     */
     type Template = TemplateStrings | TemplateStrings[];
 
-    // Fig attempts to resolve global specs first in Fig's cloud of public + private specs,
-    // while local specs are resolved locally on the user's device as a path.
+    /**
+     *
+     * The SpecLocation object defines well... the location of the completion spec we want to load.
+     * Specs can be "global" (ie hosted by Fig's cloud) or "local" (ie stored on your local machine)
+     *
+     * Global SpecLocation Object
+     * Load specs hosted in Fig's Cloud. Assume the current working directory is here: https://github.com/withfig/autocomplete/tree/master/src. Now set the value for the "name" prop to the relative location of your spec (without the .js file extension)
+     * e.g.
+     * ```js
+     * { type: "global"; name: "aws/s3" } // Loads up the aws s3 completion spec
+     * { type: "global"; name: python/http.server } // Loads up the http.server completion spec
+     * ```
```

Comment:
Does this mean that all `@example` should be replaced by this syntax?

Reply:
We can do it in a future PR